### DISCUSSION TITLE
How to use docker-compose build args in fish?

### DISCUSSION THREAD

**Discussion by wdfinch** [fish-shell/fish-shell#8825] [Q&A] [answered]
I'm really enjoying using fish but there's one command I can't seem to run in fish. I'm trying to run the following in fish:

`docker-compose build --build-arg MY_BUILD_ARG=$SOME_VAR some-service`

When I do this I get the output:

`ERROR: No such service: MY_BUILD_ARG=some_var_value`

I've tried wrapping `MY_BUILD_ARG=$SOME_VAR` in double-quotes which starts the build process. However, it seems that the variable fails to be substituted correctly and in turn, the build fails. 

Any help on how I can do this?

**Comment by krobelus** [comment, marked as answer]
> `docker-compose build --build-arg MY_BUILD_ARG=$SOME_VAR some-service`
> `ERROR: No such service: MY_BUILD_ARG=some_var_value`

You should have included how you set SOME_VAR
In this case you're probably running into Cartesian Product expansion, see https://fishshell.com/docs/current/language.html#combining-lists-cartesian-product
Note that you can always run your command with `echo ` prepended, to check the commandline after variable expansion.

**Reply by wdfinch** [reply]
Thanks for the reply. 

I was using the following to set the variable:

`set -x SOME_VAR (cat ~/someFile)`

So after some deep digging, I found out that my variable had a new line in it, and setting it this way was removing the new lines. I looked at the docs and changed it to:

`set -x SOME_VAR "$(cat ~/someFile)"`

This preserved the new lines and solved the issue. I also needed to add a double quote around `$SOME_VAR` when using it in the docker commands as well. That resolved the invalid command issue.

Thank you for your reply, it got me thinking about how I was setting the variable and new line issues. I really appreciate it.
